### ISSUE TITLE
AL0603 -  SubPageLink page property

### ISSUE THREAD

**Issue by andreilungu** [microsoft/AL#6832]
Please include the following with each issue:

**1. Describe the bug**
When use subpage part in a page and have Enum in the Subpage Link we receive error:

An implicit conversion is being performed from a value of type 'Integer' to a value of type 'Enum "Default Dimension Parent Type"'. This conversion can lead to unexpected runtime issues. This warning will become an error in a future release.AL(AL0603)

 part(defaultDimensions; "APIV1 - Default Dimensions")
                {
                    Caption = 'Default Dimensions';
                    EntityName = 'defaultDimension';
                    EntitySetName = 'defaultDimensions';
                    SubPageLink = ParentId = Field(SystemId), "Parent Type" = const(1);
                }

"Parent Type" = Enum "Default Dimension Parent Type"

How can we solve this issue considering we can not use "Parent Type" .AsInteger() or use the Enum Value instead of const(1).

**Comment by dnpb** [comment]
Why do you think you can't use the enum name in the `SubPageLink`? It should be perfectly possible. What did you try? What problem did you see?

**Comment by andreilungu** [comment]
Oh, it was just as simple as:  SubPageLink = ParentId = Field(SystemId), "Parent Type" = const(Customer);

I tried in other ways, but did not check this one.. Seems to work. Can be closed, sorry

**Comment by dnpb** [comment]
Yes, and if the enum value name contains spaces, just enclose it all in `""`, e.g. `"Parent Type" = const("Some Enum Value Name")`

**Comment by andreilungu** [comment]
What about this case?

in an XML Port we have:

`tableelement("Reservation Entry"; "Reservation Entry")
                        {
                            LinkFields = "Source Subtype" = FIELD("Document Type"), "Source ID" = FIELD("Document No."), "Source Ref. No." = FIELD("Line No.");
                            LinkTable = "Sales Line";
                            MinOccurs = Zero;
                            XmlName = 'reservationentry';
                            UseTemporary = true;
                            fieldattribute(sourcetype; "Reservation Entry"."Source Type")
                            {
                            }
}`

"Source Subtype" in table "Reservation Entry" is still an option.

"Document Type" is Enum "Sales Document Type".

So I get warning:
An implicit conversion is being performed from a value of type 'Enum "Sales Document Type"' to a value of type 'Option'. This conversion can lead to unexpected runtime issues. This warning will become an error in a future release.AL(AL0603)

**Comment by kalberes** [comment]
"Will the "Source Subtype" be converted to Enum?" Don't know transferring to app teams. They may know
